service/storage_service.cc

```diff
@@ -4293,6 +4293,36 @@ future<> storage_service::raft_removenode(locator::host_id host_id, locator::hos
     rtlogger.info("Removenode succeeded. Request ID: {}", request_id);
 }
 
+future<> storage_service::mark_excluded(const std::vector<locator::host_id>& hosts) {
+    while (true) {
+        auto guard = co_await _group0->client().start_operation(_group0_as, raft_timeout{});
+
+        std::unordered_set<raft::server_id> raft_hosts;
+        for (auto host : hosts) {
+            if (_gossiper.is_alive(host)) {
```

Comment:
Should we also validate the the host is known?
is_alive will return false for a random host-id...